### ISSUE TITLE
Is the gradient penalty loss problematic when input image is large?

### ISSUE THREAD

**Issue by shaform** [igul222/improved_wgan_training#50]
Hi,
Here the sum of squares is computed
`slopes = tf.sqrt(tf.reduce_sum(tf.square(gradients), reduction_indices=[1]))`

However, when the input image is extremely large, the dimension of gradients would be huge. It seems possible that the resulting slopes would be extremely large compared to 1.
Is this the case?

**Comment by igul222** [comment]
In theory I agree with you; in practice I haven't found compelling evidence that it's a problem. If you find any I'd love to know :)

**Comment by martinarjovsky** [comment]
You might want to experiment with replacing the penalty of (norm(grad) - 1)^2 by (norm(grad) - k)^2 for a larger k > 1.

Let us know if you find anything interesting :)
